rust/core-lib/src/editor.rs

```diff
@@ -1058,6 +1058,17 @@ impl<W: Write + Send + 'static> Editor<W> {
         Some(text.slice_to_string(offset, end_off))
     }
 
+    pub fn plugin_get_selections(&self, view_id: &ViewIdentifier) -> Value {
+        //TODO: multiview support
```

Comment:
I share your disquiet. I think the first problem is very tractable; I think we should be able to remove that inconsistency when we actually add multiview support. There was a PR a while ago from @cbrewster that included this change.

The second issue is one I struggle with, more. Part of this feels like a relic of the fact that historically there's been a 1-1 relationship between views and editors, and so they have been used interchangeably. A possible architectural change might be to actually have the main `Documents` object store views, instead of editors?

I'm very open to anybody's thoughts on this.